news_events/etl/utils.py

```diff
@@ -131,3 +132,69 @@ def parse_date(text_date: str) -> datetime:
         except:  # noqa: E722
             logging.exception("unparsable date received - ignoring '%s'", text_date)
     return dt_utc
+
+
+def convert_to_utc(dt: datetime, known_tz: str) -> datetime:
+    """
+    Convert a datetime object to UTC timezone. If its
+    orignal timezone is not known, assume it is in US/Eastern.
+
+    Args:
+        dt (datetime): The datetime object to convert
+        known_tz (str): The timezone string if known
+
+    Returns:
+        datetime: The datetime object in UTC timezone
+    """
+    if not dt:
+        return None
+    if not known_tz:
+        # Assume it is in US/Eastern where MIT is
+        dt = dt.replace(tzinfo=ZoneInfo("US/Eastern"))
+    return dt.astimezone(UTC)
+
+
+def parse_date_time_range(
```

Comment:
This does not seem to get the correct time 100% of the time. For example

start_date_str="2020-01-28"
end_date_str="2020-01-28"
time_range_str="2-3:00pm"
returns start_date = datetime.datetime(2020, 1, 28, 2, 0)
end_date = datetime.datetime(2020, 1, 28, 15, 0) 

before the conversion to utc

Also some events just have a start date like "1pm" and the time isn't parsed at all in that case

Reply:
Some events also have time strings that are going to be really hard to parse like "8:00pm; doors open at 7:30pm" I wonder if it makes sense to give up, store event dates then store event time as a string so it's displayed the way it was in the original listing

Reply:
Maybe I was trying too hard to get the times right; what's most important is really the start date because that's the only thing that's actually shown on the frontend.  Next most important is end_date, because both are used to determine if the event should be published (start date or end date should be < now).  If the professional ed api is sending back dates like "1 pm", that's just bad data and on them to fix.

How about I make sure the dates are correctly parsed (assuming the data is good), then for times, if it can't figure it out, it will just default to noon?  That way when the pipeline runs in the early AM, it will keep events happening that day published for display.